src/diff/index.js

```diff
@@ -499,6 +498,22 @@ function diffElementNodes(
 			}
 
 			newVNode._children = [];
+		} else if (typeof newChildren === 'string') {
+			if (newChildren !== oldProps.children) {
+				while (oldVNode._children && (i = oldVNode._children.pop())) {
+					// Setting textContent on the dom element will unmount all DOM nodes
+					// of the previous children, so we don't need to remove DOM in this
+					// call to unmount
+					unmount(i, oldVNode, true);
+				}
+
+				newVNode._children = [
+					// @ts-expect-error
+					createVNode(null, (dom.textContent = newChildren), null, null)
```

Comment:
We might want to consider using `dom.appendChild(document.createTextNode(newChildren))` and manually removing previous children. Otherwise, rendering a single text child will blow away non-preact-rendered children from a parent element.